pkg/k8s/aggregator.go

```diff
@@ -83,21 +89,19 @@ func NewIngressAggregator(sources []Ingresswatcher) *IngressAggregator {
 func (i *IngressAggregator) List() ([]v1.Ingress, error) {
 	is := make([]v1.Ingress, 0)
 	for _, store := range i.stores {
-		ingresses := store.List()
+		ingresses := store.cachestore.List()
 		for _, obj := range ingresses {
 			ingress, ok := obj.(*v1.Ingress)
 			if !ok {
 				return nil, fmt.Errorf("unexpected object in store: %+v", obj)
 			}
 			// check if loadbalancer status exist in ingress
 			if len(ingress.Status.LoadBalancer.Ingress) <= 0 {
-				if len(i.IngressEndpoints) > 0 {
```

Comment:
Bear with me, I'm not sure I completely follow the logic here - it seems that earlier we looked at `IngressAggregator.IngressEndpoints` string array to figure out endpoints, but now that doesn't seem to be used for anything, and instead we look at the `Store.endpoints` string array, which is defined per-source, right?

Trying to map that into our reality, is it then so that in the earlier implementation all the IngressEndpoints for all our clusters would be in `IngressAggregator.IngressEndpoints`, and one would randomly be picked? And in the new implementation, the endpoints are tied to a specific Store, which actually contains an Ingress and the endpoints associated with that ingress? No, that's not making sense to me.

ANYWAY - assuming that `IngressAggregator.IngressEndpoints` isn't used anywhere anymore, I think we should remove it from muddying up the waters. And maybe also rename `Store.endpoints` to `Store.IngressEndpoints` to conform with other occurrences of the same concept.

Reply:
I am agreed with you i removed `IngressAggregator.IngressEndpoints` from `IngressAggregator` to avoid further misunderstanding.
and you realized it very well yeah previously it was a pool of endpoints from all clusters stored in `IngressAggregator.IngressEndpoints` and picked randomly and set as `loadbalancer` status `ip` for each ingress who doesn't have any `status`. 
but the logic had some issues 
* we don't want to pick random IP from a list of all available cluster we want to pick `ips` who are associated with actual ingress in the specific cluster that ingress exist 
* The previous logic only allows us to pick single IP but in the production environment we need to have loadbalance between actual node ips